hammers/scripts/clean_old_aggregates.py

```diff
@@ -63,21 +77,75 @@ def clear_aggregates(agg_list):
+    hosts_from_aggs = []
+    for agg in allaggs.values():
+        for host in agg['hosts']:
+            hosts_from_aggs.append(host)
+
+    # Make list of ironic hosts not in any aggregate
+    ironic_hosts = osrest.ironic.nodes(auth, details=False).keys()
+    orphans = []
+    for host in ironic_hosts:
+        if host not in hosts_from_aggs:
+            host_id = [h['id'] for h in osrest.blazar.hosts(auth).values() if h['uid'] == host][0]
+            orphans.append(host_id)
+    return(orphans)
+
+def has_active_allocation(orph):
+
+    matching_allocs = [alloc for alloc in host_allocs if alloc['resource_id'] == orph]
+    if not matching_allocs:
+        return False
+    res = matching_allocs[0]['reservations'][0]['id']
+    return(res)
+
+def del_expired_alloc(db, old_alloc):
+
+    ha_id = old_alloc['id']
+    l_id = old_alloc['lid']
+    hh = old_alloc['hypervisor_hostname']
+
+    msg=("Deleting host_allocation for host {} matching expired lease {}. ".format(hh,l_id) + "\n")
+    count = query.blazar_old_host_alloc_delete(db, ha_id)
+    return(msg)
 
 def main(argv=None):
+    
     slack = Slackbot(args.slack, script_name='clean-old-aggregates') if args.slack else None
 
     try:
         term_leases = [lease for lease in leases.values() if is_terminated(lease)]
         old_aggregates = [aggs for aggs in (aggregates_for_lease(lease) for lease in term_leases) if aggs != None]
         aggregate_list = list(itertools.chain(*old_aggregates))
-        errors, report = clear_aggregates(aggregate_list)
-
-        if report:
-            str_report = ''.join(report)
-
-            print(str_report)
-
+        errors, reports = clear_aggregates(aggregate_list)
+        orphan_list = orphan_find(aggregates)
+        reports = []
+        
+        if orphan_list:
+            for orphan in orphan_list:
+                destiny = has_active_allocation(orphan)
+                host = osrest.blazar.host(auth, orphan)
+                if destiny is None:
+                    reports.append("Error identifying allocation for orphan host {}.".format(orphan) + "\n")
+                elif destiny is False:
+                    reports.append("Returning orphan host {} to freepool.".format(orphan) + "\n")
+                    osrest.nova.aggregate_add_host(auth, 1, host['hypervisor_hostname'])
+                else:
+                    destination_agg = [aggr['id'] for aggr in aggregates.values() if aggr['name'] == destiny][0]
+                    reports.append("Moving orphan host {} to destined aggregate {}.".format(orphan, destination_agg) + "\n")
+                    osrest.nova.aggregate_add_host(auth, destination_agg, host['hypervisor_hostname'])
+        
+
+        old_allocations = query.blazar_find_old_host_alloc(conn)
+        for alloc in old_allocations:
+            reports.append(del_expired_alloc(conn, alloc))
+        conn.db.commit()
+
+        if reports:
+            str_report = ''.join(reports)
```

Comment:
One trick I do is 
```suggestion
            str_report = '\n'.join(reports)
```
... which turns each line into its own line.